src/groupByKey.ts

```diff
@@ -2,6 +2,18 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: string): GroupsMap<T> {
+  const map: GroupsMap<T> = {};
+
+  for (const item of items) {
+    const groupKey = item[key];
+
+    if (map[groupKey] === undefined) {
```

Comment:
The check for `undefined` can be simplified by using the logical OR assignment operator (||=) introduced in ES2021, which will make the code more concise and readable.